src/log/Entry.h

```diff
@@ -27,7 +30,9 @@ class Entry {
     m_thread(pthread_self()),
     m_prio(pr),
     m_subsys(sub)
-  {}
+  {
+    m_thread_name = Thread::fetch_thread_name(m_thread);
```

Comment:
Ouch, I think I see why there is an interest in spinlocks.
How frequently is this called? On every log entry creation?

If the performance is a concern, please note `m_thread` here is just `pthread_self()`. This gives some optimization opportunities. Maybe instead of this complex, lock-protected container every `Thread` could cache its name in a `thread_local` variable from which the `Entry` constructor would simply copy?

Reply:
* @rzarzynski thanks for pointing me to `ceph::spinlock`
* Yeah, `Thread::fetch_thread_name(thread_id)` is called on every log entry creation
* The very issue I'm trying to solve is a thread disappearing by the time `Log::dump_recent()` is called and `Log::dump_recent()` invoking `pthread_getname_np()` with a thread id that isn't valid anymore ... so saving anything in a `thread_local` variable/storage isn't helpful

Reply:
In this exact line you, instead of going to the new registry, you could have simply called `ceph_pthread_getname()`.
I guess you aren't doing it because of being afraid about costs (BTW: this presumption might be worth checking on its own; some syscalls are executed entirely in userpsace thankfully to VDSO).
If so, the registry is just an optimization. My point is this optimization can be pretty costly in heavily multi-cored system while the standardized solution is available and TLS typically is lock free. Please note two things:

1. `m_thread_name = Thread::fetch_thread_name(m_thread)` is executed with `m_thread` set to the return of `pthread_self()` – you're obtaining the name of thread that executes this code.
2.  You already don't care about thread name changes.

Reply:
> I guess you aren't doing it because of being afraid about costs (BTW: this presumption might be worth checking on its own; some syscalls are executed entirely in userpsace thankfully to VDSO).
> If so, the registry is just an optimization. My point is this optimization can be pretty costly in heavily multi-cored system while the standardized solution is available and TLS typically is lock free. Please note two things:

Yes. @mchangir and me discussed this last week. Should be do some tests b/w pthread_getname calls and trying to fetch from a container?